chore(table): update table tutorial to be more easily understandable (#224)
patternfly/training-scenarios: react-components/table-intro/step9.md

> In this step, we'll add in the functionality of paginating the table's dataset. We'll add a new state property to track the dynamic rows and add a couple more handler functions to facilitate keeping the table and pagination in sync.

Reviewer: ```suggestion
Add functionality for paginating the table's dataset. This includes adding a new state property to track the dynamic rows and adding a few more handler functions to facilitate keeping the table and pagination in sync.
```

> 1) <strong>Create a new state property to keep track of our dynamic rows.</strong>

Reviewer: ```suggestion
1) <strong>Create a new state property to keep track of dynamic rows.</strong>
```

> Add the following code just below the other state properties that were added in step 8. Set the initial value to the `defaultRows` constant imported from `data.js`.

Reviewer 2: ```suggestion
Add the following code just below the other state properties added in step 8. Set the initial value to the `defaultRows` constant imported from `data.js`.
```

> Add the following code just below the other state properties you added previously. Set the initial value to the `defaultRows` constant imported from data.js.

Reviewer: when you say added previously - refer to the step
wrap data.js in back-ticks

> Add the following code just below the other state properties you added previously. Set the initial value to the `defaultRows` constant imported from data.js.

Reviewer 2: ```suggestion
Add the following code just below the other state properties that were added in step 1. Set the initial value to the `defaultRows` constant imported from `data.js`.
```

<pre class="file">
const [rows, setRows] = React.useState(defaultRows.slice(0, defaultPerPage));
</pre>

2) <strong>Update the `<Table />`'s `rows` prop to use the new `rows` state property.</strong>

> Locate the code that sets the rows for the table `rows={defaultRows}`, and update it to reference the new rows state property you just created.

Reviewer: when you say "just created" refer to the step

<pre class="file">
rows={rows}
</pre>

> <strong>Note: </strong> the `itemCount` prop for the pagination component will still reference the defaultRows constant as it needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown for a given page within the dataset.

Reviewer 2: ```suggestion
<strong>Note: </strong> the `itemCount` prop for the pagination component will still reference the defaultRows constant as it needs to be aware of the total number of rows in the entire dataset, not the number of rows that are being shown for a given page within the dataset.
```

> <strong>Note: </strong> the itemCount prop for the Pagination component will still reference the defaultRows constant as it needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown for a given page within the dataset.

Reviewer: wrap itemCount and defaultRows in back-ticks
Pagination component should be lower case

C) <strong>Create a function to handle updating the current page.</strong>

Copy the following handler function just below the `handlePerPageSelect` function.

<pre class="file">
const handleSetPage = (_evt, newPage, perPage, startIdx, endIdx) => {
  setCurrentPage(newPage);
}
</pre>

D) <strong>Set the `onSetPage` prop of the `<Pagination />` component to the `handleSetPage` handler function.</strong>

<pre class="file">
onSetPage={handleSetPage}
</pre>

E) <strong>Update the visible `rows` when `page` and `numPerPage` state variables are updated.</strong>

Copy the following statement into the `handlePerPageSelect` and `handleSetPage` function bodies.

<pre class="file">
setRows(defaultRows.slice(startIdx, endIdx));
</pre>

> The table should look something like the following:

Reviewer 2: ```suggestion
The table should look like the following:
```

<img src="table-intro/assets/step-9-complete.png" alt="Image of what table looks like at the end of step 9." style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />

> At this point, you should see basic pagination in place. Nice!

Reviewer: remove "you"

Reviewer 2: ```suggestion
At this point, the basic pagination is in place. Nice!
```
